hashintel/hash
# Create `CreateEntityModal` component

## 🌟 What is the purpose of this PR?

This PR creates `CreateEntityModal` component. It's not used anywhere for now. But it'll be used soon for providing `properties` to blocks with good UX.

⚠️ There are few other small changes on this PR, explained at  _🔍 What does this change_ section below ⚠️ 

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203543021352032/1203701168177453/f) _(internal)_
- [Asana task](https://app.asana.com/0/1203543021352041/1203756383149343/f) _(internal)_

## 🔍 What does this change?

Here is the list of small fixes on this PR

- Move sidebar specific `Drawer` config from `MUI` theme to `sidebar.tsx`
- Create a reusable `use-draft-entity-subgraph` hook 
- Fix render issue for boolean values on `value-cell`
- Move `gap: 6.5px` from `EntityPageWrapper` to `EntityEditor` (to make entity editor styling more consistent)
- Comment out `PeersSection` on entity editor temporarily

## 🐾 Next steps

- Fix backend related issues to make the edit modal actually function with BP blocks.
- Moving `EntityEditor` and `EditEntityModal` into `/components` folder

## ❓ How to test this?



1.  Checkout the branch
1. Open any page and create open the block context bar
2. Click on "Edit Block" option
1.  Confirm that `CreateEntityModal` looks and works as expected.

## 📹 Demo

https://user-images.githubusercontent.com/39553853/213407380-335f7baf-5ef0-4226-ad8e-8f96c25934bd.mov
